tests/common/devices.py

```diff
@@ -102,58 +102,103 @@ def __init__(self, ansible_adhoc, hostname):
 
 class SonicHost(AnsibleHostBase):
     """
-    @summary: Class for SONiC switch
+    A remote host running SONiC.
 
-    For running ansible module on the SONiC switch
+    This type of host contains information about the SONiC device (device info, services, etc.),
+    and also provides the ability to run Ansible modules on the SONiC device.
     """
+
+    # TODO: Because people are editing this variable in a bunch of places it should probably be
+    # revised to "DEFAULT_CRITICAL_SERVICES", and we should make critical_services a property of
+    # SonicHost
     CRITICAL_SERVICES = ["swss", "syncd", "database", "teamd", "bgp", "pmon", "lldp", "snmp"]
 
-    def __init__(self, ansible_adhoc, hostname, gather_facts=False):
+    def __init__(self, ansible_adhoc, hostname):
         AnsibleHostBase.__init__(self, ansible_adhoc, hostname)
-        if gather_facts:
-            self.gather_facts()
+        self._facts = self._gather_facts()
+        self._os_version = self._get_os_version()
+
+        if self.facts["num_npu"] > 1:
+            self._update_critical_services_for_multi_npu()
 
-    def _get_critical_services_for_multi_npu(self):
+    @property
+    def facts(self):
         """
-        Update the critical_services with the service names for multi-npu platforms
+        Platform information for this SONiC device.
+
+        Returns:
+            dict: A dictionary containing the device platform information.
+
+            For example:
+            {
+                "platform": "x86_64-arista_7050_qx32s",
+                "hwsku": "Arista-7050-QX-32S",
+                "asic_type": "broadcom",
+                "num_npu": 1
+            }
         """
-        m_service = []
-        for service in self.CRITICAL_SERVICES:
-            for npu in self.facts["num_npu"]:
-                npu_service = service+npu
-                m_service.insert(npu, npu_service)
-        self.CRITICAL_SERVICES = m_service
-        print self.CRITICAL_SERVICES
 
-    def _get_npu_info(self):
+        return self._facts
+
+    @property
+    def os_version(self):
+        """
+        The OS version running on this SONiC device.
+        
+        Returns:
+            str: The SONiC OS version (e.g. "20181130.31")
+        """
+
+        return self._os_version
+
+    def _gather_facts(self):
+        """
+        Gather facts about the platform for this SONiC device.
+        """
+
+        facts = dict()
+        facts.update(self._get_platform_info())
+        facts["num_npu"] = self._get_npu_count(facts["platform"])
```

Comment:
curious if we could make platform,.. properies of the class instead if having them be part of facts dict. Similar to os_version.

Reply:
Agree 100%, it's just they're used in quite a few places at the moment so I wanted to do a follow-up PR for that change. :)